parboiled-core/src/main/scala/org/parboiled2/Parser.scala

```diff
@@ -94,16 +94,19 @@ abstract class Parser(initialValueStackSize: Int = 16, maxValueStackSize: Int =
   private var _cursor: Int = _
 
   // the value stack instance we operate on
-  private var _valueStack: ValueStack = _
+  private var _valueStack: ValueStack = new ValueStack(initialValueStackSize, maxValueStackSize)
 
   // the current ErrorAnalysisPhase or null (in the initial run)
   private var phase: ErrorAnalysisPhase = _
 
+  private var _maxLength = -1
```

Comment:
How about `_inputLength` instead?
This var isn't really updated to hold some kind of "max" value, it seems to be a simple cache for `input.length`.

Reply:
Done and pushed.